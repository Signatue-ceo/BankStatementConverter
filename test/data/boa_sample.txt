BANK OF AMERICA
Monthly Statement
Account: ****5678
Period: 02/01/2024 - 02/29/2024

Beginning Balance: $3,250.00
Ending Balance: $3,125.50

02/01/2024  Beginning Balance              $3,250.00
02/03/2024  CHECK DEPOSIT                 +$500.00   $3,750.00
02/05/2024  WALMART PURCHASE              -$125.75   $3,624.25
02/08/2024  DIRECT DEPOSIT                +$1,800.00 $5,424.25
02/10/2024  RESTAURANT                    -$65.00    $5,359.25
02/12/2024  GAS STATION                   -$40.00    $5,319.25
02/15/2024  ONLINE SUBSCRIPTION           -$29.99    $5,289.26
02/18/2024  GROCERY STORE                 -$85.50    $5,203.76
02/20/2024  PHARMACY                      -$35.00    $5,168.76
02/22/2024  MOVIE THEATER                 -$55.00    $5,113.76
02/25/2024  COFFEE SHOP                   -$8.50     $5,105.26
02/28/2024  BANK FEE                      -$12.00    $5,093.26
