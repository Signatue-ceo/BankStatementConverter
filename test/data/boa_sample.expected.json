{
  "id": "mock-id",
  "bankName": "BANK OF AMERICA",
  "accountNumber": "****5678",
  "statementPeriod": {
    "start": "2024-02-01T05:00:00.000Z",
    "end": "2024-02-29T05:00:00.000Z"
  },
  "openingBalance": 3250,
  "closingBalance": 3125.50,
  "transactions": [
    {
      "id": "mock-id",
      "date": "2024-02-01T00:00:00.000Z",
      "description": "Beginning Balance",
      "amount": 3250,
      "type": "credit",
      "balance": 3250,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-03T00:00:00.000Z",
      "description": "CHECK DEPOSIT",
      "amount": 500,
      "type": "credit",
      "balance": 3750,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-05T00:00:00.000Z",
      "description": "WALMART PURCHASE",
      "amount": 125.75,
      "type": "debit",
      "balance": 3624.25,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-08T00:00:00.000Z",
      "description": "DIRECT DEPOSIT",
      "amount": 1800,
      "type": "credit",
      "balance": 5424.25,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-10T00:00:00.000Z",
      "description": "RESTAURANT",
      "amount": 65,
      "type": "debit",
      "balance": 5359.25,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-12T00:00:00.000Z",
      "description": "GAS STATION",
      "amount": 40,
      "type": "debit",
      "balance": 5319.25,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-15T00:00:00.000Z",
      "description": "ONLINE SUBSCRIPTION",
      "amount": 29.99,
      "type": "debit",
      "balance": 5289.26,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-18T00:00:00.000Z",
      "description": "GROCERY STORE",
      "amount": 85.50,
      "type": "debit",
      "balance": 5203.76,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-20T00:00:00.000Z",
      "description": "PHARMACY",
      "amount": 35,
      "type": "debit",
      "balance": 5168.76,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-22T00:00:00.000Z",
      "description": "MOVIE THEATER",
      "amount": 55,
      "type": "debit",
      "balance": 5113.76,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-25T00:00:00.000Z",
      "description": "COFFEE SHOP",
      "amount": 8.50,
      "type": "debit",
      "balance": 5105.26,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-02-28T00:00:00.000Z",
      "description": "BANK FEE",
      "amount": 12,
      "type": "debit",
      "balance": 5093.26,
      "reference": null
    }
  ],
  "metadata": {
    "processingTime": 0,
    "confidence": 0,
    "sourceFormat": "pdf",
    "totalTransactions": 12
  }
}